src/test/java/commandparser/StringParserTest.java

```diff
@@ -0,0 +1,165 @@
+package commandparser;
+
+import data.exceptions.TaskManagerException;
+import org.junit.jupiter.api.Test;
+import time.MonthView;
+import time.WeekView;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+
+class StringParserTest {
```

Comment:
Thanks for adding comprehensive testing to the newly implemented StringParser class!